### PR TITLE
Make default EventLoop be allocated on stack

### PR DESCRIPTION
**Pull request by FolMing**

### CODE REVIEW

File: stout/event-loop.cc [hunk 1]
```diff
@@ -79,20 +79,34 @@ void EventLoop::Clock::Advance(const std::chrono::nanoseconds& nanoseconds) {
...
-EventLoop& EventLoop::Default(EventLoop* replacement) {
-  delete CHECK_NOTNULL(loop);
-  loop = CHECK_NOTNULL(replacement);
-  return Default();
+void EventLoop::DefaultReset() {
+  if (loop_created) {
+    loop->~EventLoop();
+    for (int i = 0; i < sizeof(EventLoop); i++) {
+      loop_memory[i] = 0;
+    }
```
Comment (rjhuijsman) [inline stout/event-loop.cc:104]: As mentioned by Gorm on the call, [`memset`](https://www.cplusplus.com/reference/cstring/memset/) is probably the right tool to zero out memory.

However, do we need to zero out this memory? Won't the constructor for the new `EventLoop` overwrite it anyway?

File: stout/event-loop.h [hunk 2]
```diff
@@ -202,7 +202,7 @@ class EventLoop : public Scheduler {
   //
   // NOTE: takes ownership of event loop pointer.
```
Comment (rjhuijsman) [inline stout/event-loop.h:203]: This comment is out of date now that `DefaultReset` doesn't take a pointer anymore.

**Review by rjhuijsman — APPROVED** [review]
Having thought about this, this is safe because the memory we're allocating it in is _static_ (it's not actually on the stack where it will get destroyed at some point, but alive for the lifetime of the program).

Approved! Please consider the two comments before merging.

File: stout/event-loop.cc (hunk 1 above)
Comment (FolMing) [reply stout/event-loop.cc]: `new EventLoop` - will not overwrite it.
`new EventLoop()` - will.

Changed the code to use the second one and removed that `for` loop.

File: stout/event-loop.h (hunk 2 above)
Comment (FolMing) [reply stout/event-loop.h]: Fixed!

**Pull request merged**